flepimop/gempyor_pkg/src/gempyor/shared_cli.py

```diff
@@ -0,0 +1,213 @@
+        ["-i", "--first_sim_index"],
+        envvar="FIRST_SIM_INDEX",
+        type=click.IntRange(min=1),
+        default=1,
+        show_default=True,
+        help="The index of the first simulation",
+    ),
+    "stoch_traj_flag": click.Option(
+        ["--stochastic/--non-stochastic", "stoch_traj_flag"],
+        envvar="FLEPI_STOCHASTIC_RUN",
+        default=False,
+        help="Run stochastic simulations?",
+    ),
+    "write_csv": click.Option(
+        ["--write-csv/--no-write-csv"],
+        default=False,
+        show_default=True,
+        help="write csv output?",
+    ),
+    "write_parquet": click.Option(
+        ["--write-parquet/--no-write-parquet"],
+        default=True,
+        show_default=True,
+        help="write parquet output?",
+    ),
+}
+
+
+# adapted from https://stackoverflow.com/a/78533451
+def click_helpstring(params: Union[click.Parameter, List[click.Parameter]]):
+    """
+    A decorator that dynamically appends `click.Parameter`s to the docstring of the decorated function.
+
+    Args:
+        params (Union[click.Parameter, List[click.Parameter]]): A parameter or a list of parameters whose corresponding argument values and help strings will be appended to the docstring of the decorated function.
+
+    Returns:
+        Callable: The original function with an updated docstring.
+    """
+    if not isinstance(params, list):
+        params = [params]
+
+    def decorator(func):
+        # Generate the additional docstring with args from the specified functions
+        additional_doc = "\n\nCommand Line Interface arguments:\n"
+        for param in params:
+            paraminfo = param.to_info_dict()
+            additional_doc += f"\n{paraminfo['name']}: {paraminfo['type']['param_type']}\n"
+
+        if func.__doc__ is None:
+            func.__doc__ = ""
+        func.__doc__ += additional_doc
+
+        return func
+
+    return decorator
+
+# TODO: create a custom command decorator cls ala: https://click.palletsprojects.com/en/8.1.x/advanced/#command-aliases
+# to also apply the `@click_helpstring` decorator to the command. Possibly to also default the params argument, assuming
+# enough commands have consistent option set?
+
+@click_helpstring([config_files_argument] + list(config_file_options.values()))
+def parse_config_files(
+    config_files: list[pathlib.Path],
+    config_filepath: pathlib.Path = None,
+    seir_modifiers_scenarios: list[str] = [],
+    outcome_modifiers_scenarios: list[str] = [],
+    in_run_id: str = None,
+    out_run_id: str = None,
+    in_prefix: str = None,
+    nslots: int = None,
+    jobs: int = None,
+    write_csv: bool = False,
+    write_parquet: bool = True,
+    first_sim_index: int = 1,
+    stoch_traj_flag: bool = False,
+) -> None:
+    """
+    Parse configuration file(s) and override with command line arguments
+
+    Args: (see auto generated CLI items below)
+
+    Returns: None (side effect: updates the global configuration object)
+    """
```

Comment:
The help page:
```
Help on function parse_config_files in module gempyor.shared_cli:

parse_config_files(config_files: list[pathlib.Path], config_filepath: pathlib.Path = None, seir_modifiers_scenarios: list[str] = [], outcome_modifiers_scenarios: list[str] = [], in_run_id: str = None, out_run_id: str = None, in_prefix: str = None, nslots: int = None, jobs: int = None, write_csv: bool = False, write_parquet: bool = True, first_sim_index: int = 1, stoch_traj_flag: bool = False) -> None
        Parse configuration file(s) and override with command line arguments

        Args: (see auto generated CLI items below)

        Returns: None (side effect: updates the global configuration object)


    Command Line Interface arguments:

    config_files: Path

    config_filepath: Path

    seir_modifiers_scenarios: String

    outcome_modifiers_scenarios: String

    jobs: IntRange

    nslots: IntRange

    in_run_id: String

    out_run_id: String

    in_prefix: String

    first_sim_index: IntRange

    stoch_traj_flag: Bool

    write_csv: Bool

    write_parquet: Bool
```
doesn't quite look like what I thought we discussed and also isn't in valid format? I thought the proposed solution was to use `**kwargs` and then list out the fields that this function would look for in a `Notes:` section or similar? 

Also, looks like missing `| None` from some of these type hints.